Source/Particles/Collision/BackgroundMCCCollision.cpp

```diff
@@ -27,8 +27,41 @@ BackgroundMCCCollision::BackgroundMCCCollision (std::string const collision_name
 
     amrex::ParmParse pp_collision_name(collision_name);
 
-    queryWithParser(pp_collision_name, "background_density", m_background_density);
-    queryWithParser(pp_collision_name, "background_temperature", m_background_temperature);
+    amrex::Real background_density = 0;
+    if (queryWithParser(pp_collision_name, "background_density", background_density)) {
+        m_background_density_parser = makeParser(std::to_string(background_density), {"x", "y", "z", "t"});
```

Comment:
Is there a performance penalty to always use the parser? Versus having a flag in the particle loop selecting between a constant value and parsed value.

Reply:
Also, should a check be added that the specified density is > 0?

Reply:
I just asked about the performance difference in the "amrex" Slack channel and Andrew said in tests he has done he saw no performance difference, so we're probably fine always using the parser.
I'm happy to add the density > 0 check. Thank for the suggestion!

Reply:
I ran some simulations, using the 1D capacitive discharge example, to check the performance impact of this PR while using constant values for the background pressure and temperature. The timings below compare this branch with the current `development` branch. On CPU I used 125000 particles per process (running 8 processes). I also ran single GPU simulations on Perlmutter using 2.5 million particles. The times shown are the inclusive max times (seconds) for `MultiParticleContainer::doCollisions()` to complete 8000 steps.
```
    development     this PR
CPU     27.28       28.87
GPU     3.422       3.463
```
There does seem to be a slight slow down on CPU although these are one off runs and the times could be slightly impacted by other processes on the machine.